api/src/s3/s3.service.ts

```diff
@@ -1,140 +1,154 @@
 import { Injectable } from "@nestjs/common";
-import { ConfigService } from "@nestjs/config";
 import * as Minio from "minio";
-import { S3Config } from "../configuration";
+import { S3CredentialDto } from "../dto/S3CredentialDto";
 
 @Injectable()
 export class S3Service {
-    client: Minio.Client;
-    s3Config: S3Config;
-
-    constructor(private configService: ConfigService) {
-        this.s3Config = this.configService.get<S3Config>("s3");
-
-        this.client = new Minio.Client({
-            endPoint: this.s3Config.endpoint,
-            port: this.s3Config.port,
-            useSSL: this.s3Config.useSSL,
-            accessKey: this.s3Config.accessKey,
-            secretKey: this.s3Config.secretKey,
-        });
-    }
+    // Note: No default client or config - everything is bucket-specific now
 
-    /**
-     * Get the configured image bucket name
-     */
-    public get imageBucket() {
-        return this.s3Config.imageBucket;
+    constructor() {
```

Comment:
Pass the S3 bucket ID to the constructor
```suggestion
    private endPoint: string;
    private port: number;
    private useSSL: boolean;
    private accessKey: string;
    private secretKey: string;
    
    constructor(bucketID: string) {
       // Logic to fetch credentials from database
       
```